src/pages/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,155 @@
+import { useSearchParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { PeopleFilters } from '../../components/PeopleFilter/PeopleFilters';
+import { Loader } from '../../components/Loader';
+import { PeopleTable } from '../../components/PeopleTable';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>();
+  const [filteredPeople, setFilteredPeople] = useState<Person[]>([]);
+  const [noPeopleOnServer, setNoPeopleOnServer] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isLoader, setIsLoader] = useState(true);
+  const [searchParams] = useSearchParams();
+
+  const filterPeople = (
+    sex: string | null,
+    query: string | null,
+    centuries: number[] | null,
+    peopleList: Person[],
+  ) => {
+    let newFilteredPeople = [...peopleList];
+
+    if (sex) {
+      newFilteredPeople = newFilteredPeople
+        .filter(person => person.sex === sex);
+    }
+
+    if (query) {
+      newFilteredPeople = newFilteredPeople.filter(person => person.name
+        .toLowerCase().includes(query));
+    }
+
+    if (centuries?.length) {
+      newFilteredPeople = newFilteredPeople.filter(
+        person => centuries.includes(Math.ceil(person.born / 100)),
+      );
+    }
+
+    return newFilteredPeople;
+  };
+
+  const sortPeople = (
+    sort: string | null,
+    order: string | null,
+    peopleList: Person[],
+  ) => {
+    const sortedPeople = [...peopleList];
+
+    switch (sort) {
+      case 'name':
+        sortedPeople.sort((a, b) => a.name.toLowerCase()
+          .localeCompare(b.name.toLowerCase()));
+        break;
+
+      case 'sex':
+        sortedPeople.sort(
+          (a, b) => a.sex.localeCompare(b.sex),
+        );
+        break;
+
+      case 'born':
+        sortedPeople.sort((a, b) => a.born - b.born);
+        break;
+
+      case 'died':
+        sortedPeople.sort((a, b) => a.died - b.died);
+        break;
```

Comment:
Combine logic with helper function 
![image](https://github.com/mate-academy/react_people-table-advanced/assets/70009205/6384c167-c4a3-4643-bcec-400885849649)
And use [computed properties](https://www.javascripttutorial.net/es6/javascript-computed-property/)

Reply:
done